python/cudf/cudf/core/column_accessor.py

```diff
@@ -341,6 +341,84 @@ def _pad_key(self, key: Any, pad_value="") -> Any:
             key = (key,)
         return key + (pad_value,) * (self.nlevels - len(key))
 
+    def replace_level_values(
+        self, mapper: Union[Mapping[Any, Any], Callable], level: int
+    ) -> ColumnAccessor:
+        """
+        Parameters
+        ----------
+        self : ColumnAccessor of a given dataframe
+
+        mapper : dict-like or function transformations to apply to
+            the column label values depending on selected ``level``.
+
+            If dict-like, only replace the specified level of the
+            ColumnAccessor's keys (that match the mapper's keys) with
+            mapper's values
+
+            If callable, the function is applied only to the specified level
+            of the ColumnAccessor's keys.
+
+        level : int
+            In case of RangeIndex, only supported level is [0, None].
+            In case of a MultiColumn, only the column labels in the specified
+            level of the ColumnAccessor's keys will be transformed.
+
+        Returns
+        -------
+        A new ColumnAccessor with values in the keys replaced according
+        to the given mapper and level.
+
+        """
+        col_names, columns = list(self.keys()), list(self.values())
+
+        if self.multiindex:
+            if isinstance(mapper, Mapping):
+                cols_list = [list(col) for col in col_names]
+                new_names = []
+                for col_name in cols_list:
+                    col_name[level] = mapper.get(
+                        col_name[level], col_name[level]
+                    )
+                    new_names.append(tuple(col_name))
```

Comment:
We should use a list comprehension here instead of initializing an empty list and then appending to it.